distributed/scheduler.py

```diff
@@ -7323,7 +7342,7 @@ def request_remove_replicas(
         )
 
 
-def _remove_from_processing(state: SchedulerState, ts: TaskState) -> str:
+def _remove_from_processing(state: SchedulerState, ts: TaskState) -> str | None:
```

Comment:
After putting some more thought into it, I think that `_remove_from_processing` should always return `None`. Its job is to 
> Remove *ts* from the set of processing tasks.

At either return statement, it has _successfully_ completed its task. 
In your PR, `transition_processing_released` is the only place where we care about the undocumented feature that it makes a distinction for us between returning `None` if it encounters a stale worker state and `ws.address` if it encounters a current worker state. IMO, this muddles up mutation and querying logic and introduces additional complexity that would at least need to be documented. Why not let `transition_processing_released` worry about it?

From #6392 we also know that returning an address is most likely not a good idea since it's not a unique identifier.

Reply:
I'd be fine with that. The one argument for it not always returning None is that it's kinda like a pop—it does `ts.processing_on = None`, so after it's run, callers can't find out what worker the task was processing on.

It could be reasonable to say that callers need to pull out `ts.processing_on` first, but then they also need to check if that WorkerState is stale, which is more places to get things wrong. Since `_remove_from_processing` has to do the stale check anyway, I kinda like having it return the result of that check (in that it returns None if the worker is stale). Returning a WorkerState instead of an address in the non-stale case might be more practical though.